fgspectra/models.py

```diff
@@ -1,89 +1,134 @@
-# -*- coding: utf-8 -*-
 r"""
-Foreground models.
+Models of cross-spectra
 
 This module draws inspiration from FGBuster (Davide Poletti and Josquin Errard)
 and BeFoRe (David Alonso and Ben Thorne).
 """
-import fgspectra.frequency, fgspectra.power
+from abc import ABC, abstractmethod
 import numpy as np
+from . import frequency as fgf
+from . import power as fgp
 
 
-class CompositeModel:
-    def __init__(self):
-        self.params = {}
-        return
+class CrossSpectrum(ABC):
+    """Base class for cross-spectra."""
 
-    def model(self, nu_i, nu_j, ell, **kwargs):
-        """Compute the model at frequency and ell combinations.
+    @abstractmethod
+    def __call__(self, *args):
```

Comment:
We should make a decision about the API.
About this one. PRO: argument order is always the same. CONS: can't fix parameters (left to the user)

fgbuster-like mechanism (`**kwargs` in `__init__` fix parameters, `*args` in `__call__` provides the rest). PRO: can fix parameters. CONS: `__call__`'s signature changes depending on construction; need for a (potentially slow) mechanism for tracking free and non-free params

Reply:
This is probably an aesthetic thing. What do you prefer?

Reply:
I like the second mechanism just because of the way we handle free and fixed params in BBPipe but we can probably handle it either way.

Reply:
The first is definitely potentially easier to use out of the box.

Reply:
I prefer the first one because that's what I am used to in fgbuster. @damonge expressed some appreciation for it as well.
On the other hand I agree that the second seems more natural for newbies. So I am happy to hear other's opinion.
If we decide for the latter, I think we should downgrade the classes to functions

Reply:
Yes I prefer the fgbuster style as well.